precompiles/gov/integration_test.go

```diff
@@ -475,5 +477,203 @@ var _ = Describe("Calling governance precompile from EOA", func() {
 				Expect(out.TallyResult.NoWithVeto).To(Equal("0"))
 			})
 		})
+
+		Context("proposal query", func() {
+			method := gov.GetProposalMethod
+			BeforeEach(func() {
+				callArgs.MethodName = method
+			})
+
+			It("should return a proposal", func() {
+				callArgs.Args = []interface{}{uint64(1)}
+				txArgs.GasLimit = 200_000
+
+				_, ethRes, err := s.factory.CallContractAndCheckLogs(
+					s.keyring.GetPrivKey(0),
+					txArgs,
+					callArgs,
+					passCheck,
+				)
+				Expect(err).To(BeNil(), "error while calling the smart contract: %v", err)
+
+				var out gov.ProposalOutput
+				err = s.precompile.UnpackIntoInterface(&out, method, ethRes.Ret)
+				Expect(err).To(BeNil())
+
+				// Check proposal details
+				Expect(out.Proposal.Id).To(Equal(uint64(1)))
+				Expect(out.Proposal.Status).To(Equal(uint32(v1.StatusVotingPeriod)))
+				Expect(out.Proposal.Proposer).To(Equal(s.keyring.GetAddr(0)))
+				Expect(out.Proposal.Metadata).To(Equal("ipfs://CID"))
+				Expect(out.Proposal.Title).To(Equal("test prop"))
+				Expect(out.Proposal.Summary).To(Equal("test prop"))
+				Expect(out.Proposal.Messages).To(HaveLen(1))
+				Expect(out.Proposal.Messages[0]).To(Equal("/cosmos.bank.v1beta1.MsgSend"))
+
+				// Check tally result
+				Expect(out.Proposal.FinalTallyResult.Yes).To(Equal("0"))
+				Expect(out.Proposal.FinalTallyResult.Abstain).To(Equal("0"))
+				Expect(out.Proposal.FinalTallyResult.No).To(Equal("0"))
+				Expect(out.Proposal.FinalTallyResult.NoWithVeto).To(Equal("0"))
+			})
+
+			It("should fail when proposal doesn't exist", func() {
+				callArgs.Args = []interface{}{uint64(999)}
+				txArgs.GasLimit = 200_000
+
+				_, _, err := s.factory.CallContractAndCheckLogs(
+					s.keyring.GetPrivKey(0),
+					txArgs,
+					callArgs,
+					defaultLogCheck.WithErrContains("proposal 999 doesn't exist"),
+				)
+				Expect(err).To(BeNil())
+			})
+		})
+
+		Context("proposals query", func() {
+			method := gov.GetProposalsMethod
+			BeforeEach(func() {
+				callArgs.MethodName = method
+			})
+
+			It("should return all proposals", func() {
+				callArgs.Args = []interface{}{
+					uint32(0), // StatusNil to get all proposals
+					common.Address{},
+					common.Address{},
+					query.PageRequest{
+						Limit:      10,
```

Comment:
nit: I would not use a limit of 10 when proposals are just 2 since it is a bit misleading. I would instead not use it when it is not required and add a test to check that the pagination is working